prowler/lib/check/models.py

```diff
@@ -413,7 +413,11 @@ class Check_Report:
     def __init__(self, metadata, resource=None):
         self.status = ""
         self.check_metadata = CheckMetadata.parse_raw(metadata)
-        self.resource_metadata = resource.dict() if resource else {}
+        self.resource_metadata = (
+            resource.dict()
+            if hasattr(resource, "dict")
+            else resource.to_dict() if hasattr(resource, "to_dict") else {}
+        )
```

Comment:
Should we change this in order to handle dataclass?

Reply:
Done